### ISSUE TITLE
Lost ability to merge or reject patches in Chrome and Firefox

### ISSUE THREAD

**Issue by atomrab** [periodo/periodo-client#240]
When logged in (fresh) with my ORCID, I no longer see the option in the Submitted change>Review page to merge or reject a given patch. I can add a comment, but there are no other actions available anywhere that I can see. This is the case on both Chrome and Firefox (both latest versions) on Mac IOS Sierra 10.12.6, even after clearing the cache on Chrome. On Safari on the same machine, however, "Accept changes?" options still appear.

**Comment by ptgolden** [comment]
I have the same issue. Let me look into it.

**Comment by ptgolden** [comment]
@rybesh this is strange- it works when I run the client locally and add a https://data.perio.do/ backend. It looks like, when running from https://client.perio.do/, the header `Link: </patches/247/merge>;rel="merge"` isn't being returned. I wonder if there is something going on with the server?

**Comment by ptgolden** [comment]
It also works when I open it in an incognito tab in Chrome. Something tells me there's something weird going on with authentication.

**Comment by ptgolden** [comment]
In Firefox, I deleted all site data for orcid and Periodo, and now it works

**Comment by ptgolden** [comment]
Adam, can you do me a favor and follow these instructions in Firefox:

1. Go to a patch request page where you should be able to merge
2. Open the Network tab in Developer Tools (I access it by going to Tools -> Web Developer -> Network)
3. Click the button that says "Reload the page to see detailed information"
4. Once everything is loaded, right click on one of the records and select "Save All as HAR"
5. Email that file to Ryan and me

I think there's something wonky with how ORCID credentials are getting stored. This might help us track it down.

**Comment by ptgolden** [comment]
So, the problem is your browser is sending the wrong ORCID credential to the server. I'm not sure why signing out and signing back in does not fix the problem.

Can you confirm that you can see the "Accept changes?" page in Chrome if you open the site in incognito mode? (Do this in Chrome, not Firefox)

**Comment by atomrab** [comment]
Yes — Ryan asked this yesterday, and it works in incognito mode.

**Comment by ptgolden** [comment]
OK, thanks. Can you do the following in the Firefox browser that's not working? 

1. Log out of ORCID on the Canonical data source on the client. (On the settings page for the data source)
2. Go to the "Privacy & security" page in Firefox's preferences. You can paste this in your URL bar: about:preferences#privacy
3. Click on the "Manage data" button. In that page, type in "orcid.org", and then select "Remove all shown"
4. Try logging in with your ORCID in the client again, and see if the "Accept changes" button shows up

If it doesn't work, I can take you through something else. I'm trying to pin down what this problem is, and why it cropped up.

**Comment by ptgolden** [comment]
If that doesn't work, can you create a new remote backend with the URL <https://data.perio.do/>, sign into that one, and see if the "Accept changes" button appears?

**Comment by atomrab** [comment]
Removing orcid.org from the Firefox cookies and site data section appears to have worked. My button is back there. 

I also get a button using the data.perio.do option in the previous message. But I can't use that as a fix in Chrome, apparently, because when I try to create a web-based data source with that URL, I get a pop-up error message that reads "Error saving data source".

**Comment by ptgolden** [comment]
OK, so the issue was ORCID.

Does your Chromium browser still not show the button when you log in normally? If so, please don't try to fix it-- Ryan and I want to figure out how to deal with whatever is going on with ORCID without having to clear your cookies.

I will look into the "Error saving data source" message

**Comment by atomrab** [comment]
Huh, Chrome is back now -- I logged out and in again (which I'd done several times before), and now my button has returned. Hooray, I guess?

**Comment by ptgolden** [comment]
Here is my guess as to what is going on: ORCID used to respond to authentication requests with the same token every time, and now they respond with a different one for each login.

We only store the last token as the valid one for a user, so any new login (i.e. in another browser, which doesn't share the same stored credential since it's in browser storage) invalidates any other browsers that are already authenticated.

I confirmed this just now by doing the following:
1. I logged in on Firefox with my ORCID credentials, and was recognized by the server as having sufficient privileges. That is, I was presented with an option to merge a patch.
2. I logged in in a Chromium incognito tab, and was also recognized as having sufficient privileges
3. Going back to Firefox, I no longer had sufficient privileges to merge a patch

Looking at the server logs that @rybesh added to print out all authentication events confirms that a new authentication token was issued in both (1) and (2).

(Note that this does not explain why logging out and logging back in wasn't fixing the problem before. Maybe it was @rybesh poking around that fixed this?)
